internal/status/all_replicas_ready_test.go

```diff
@@ -107,23 +108,23 @@ var _ = Describe("AllReplicasReady", func() {
 			When("transitions to true", func() {
 				BeforeEach(func() {
 					sts.Status.ReadyReplicas = 5
-					sts.Status.Replicas = 5
+					sts.Spec.Replicas = &desiredReplicas
 				})
 
 				It("updates the transition time", func() {
-					condition := rabbitmqstatus.AllReplicasReadyCondition([]runtime.Object{&corev1.Endpoints{}, sts}, existingCondition)
+					condition := rabbitmqstatus.AllReplicasReadyCondition([]runtime.Object{&corev1.Endpoints{}, sts}, oldCondition)
 					Expect(condition.LastTransitionTime).ToNot(Equal(emptyTime))
 				})
 			})
 
 			When("transitions to false", func() {
```

Comment:
I know it's not a change made by this PR but a suggestion. -
The transition to false or true does not really seem to have a previous state. The test blocks are a bit confusing. We can merge these with the `When` blocks above where another `It` could be verifying that the transition time is set. It's upto you.